beetsplug/parentwork.py

```diff
@@ -0,0 +1,159 @@
+        'performer':            types.STRING,
+        'performer_sort':       types.STRING,
+        'parent_composer':      types.STRING,
+        'parent_composer_sort': types.STRING}
+
+    def imported(self, session, task):
+        self.find_work(task.items)
+
+    def find_work(self, items):
+    
+        def find_parentwork(work_id):
+            "finds the parentwork of a work given its id"
+            work_info = musicbrainzngs.get_work_by_id(work_id,
+                includes=["work-rels", "artist-rels"])
+            partof = True
+            while partof:
+                partof = False
+                if 'work-relation-list' in work_info['work']:
+                    for work_father in work_info['work'][
+                            'work-relation-list']:
+                        if work_father['type'] == 'parts' and\
+                                'direction' in work_father:
+                            if work_father['direction'] == 'backward':
+                                father_id = work_father['work']['id']
+                                partof = True
+                                work_info = musicbrainzngs.get_work_by_id(
+                                    father_id,includes=["work-rels", 
+                                        "artist-rels"])
+            return work_info
+
+        for item in items:
+            performer            = []
+            performer_sort       = []
+            work                 = []
+            work_disambig        = []
+            parent_work          = []
+            parent_work_disambig = []
+            parent_composer      = []
+            parent_composer_sort = []
+            item.read()
+            recording_id = item['mb_trackid']
+            performer_types = ['performer', 'instrument', 'vocal',
+                'conductor', 'performing orchestra', 'chorus master', 
+                    'concertmaster']
+            i = 0
+            while i < 5:
+                try:
+                    rec_rels = musicbrainzngs.get_recording_by_id(
+                        recording_id, includes=['work-rels', 'artist-rels'])
+                    if 'artist-relation-list' in rec_rels['recording']:
+                        for dudes in rec_rels['recording'][
+                                'artist-relation-list']:
+                            if dudes['type'] in performer_types:
+                                performer.append(dudes['artist']['name'])
+                                    performer_sort.append(dudes['artist']
+                                        ['sort-name'])
+                    if 'work-relation-list' in rec_rels['recording']:
+                        for work_relation in rec_rels['recording'][
+                                'work-relation-list']:
+                            if work_relation['type'] != 'performance':
+                                continue
+                            work_id = work_relation['work']['id']
```

Comment:
Would it be possible to factor out some of the code in the middle of this long loop into its own function? Being able to see how one work relation is processed at a time, for example, could greatly help with readability.